src/utils/archiver.js

```diff
@@ -0,0 +1,35 @@
+'use strict'
+
+const { lstat } = require('fs/promises')
+const AdmZip = require('adm-zip')
+
+const archiveItem = async (path, out) => {
+  let isDirectory = false
+  try {
+    const stat = await lstat(path)
+    isDirectory = stat.isDirectory()
+  } catch (err) {
+    throw new Error(
+      'An error occurred while checking if file or directory: ' + err.message
+    )
+  }
```

Comment:
avoid this mutable state pattern. extract a function and you don't need this pattern as you can simply return the value